src/js/tech/html5.js

```diff
@@ -115,6 +120,59 @@ class Html5 extends Tech {
   }
 
   /**
+   * When a captions track is enabled in the iOS Safari native player, all other
+   * tracks are disabled (including metadata tracks), which nulls all of their
+   * associated cue points. This will restore metadata tracks to their pre-fullscreen
+   * state in those cases so that cue points are not needlessly lost.
+   *
+   * @private
+   */
+  saveAndRestoreMetadataTrackModeInIOSNativePlayer_() {
+    const textTracks = this.textTracks();
+    let metadataTracksPreFullscreenState;
+
+    // captures a snapshot of every metadata track's current state
+    const takeMetadataTrackSnapshot = () => {
+      metadataTracksPreFullscreenState = [];
+
+      for (let i = 0; i < textTracks.length; i++) {
+        const track = textTracks[i];
+
+        if (track.kind === 'metadata') {
+          metadataTracksPreFullscreenState.push({
+            track,
+            storedMode: track.mode
+          });
+        }
+      }
+    };
+
+    // snapshot each metadata track's initial state, and update the snapshot
+    // each time there is a track 'change' event
+    takeMetadataTrackSnapshot();
+    textTracks.on('change', takeMetadataTrackSnapshot);
+
+    // when we enter fullscreen playback, restore all track modes to their initial state
+    this.on('fullscreenchange', (event, data) => {
+      if (data.isFullscreen) {
+        // stop updating the snapshot now that we are in fullscreen
+        textTracks.off('change', takeMetadataTrackSnapshot);
+
+        textTracks.one('change', () => {
+          for (let i = 0; i < metadataTracksPreFullscreenState.length; i++) {
+            const storedTrack = metadataTracksPreFullscreenState[i];
+
+            storedTrack.track.mode = storedTrack.storedMode;
```

Comment:
I think I should wrap this in a `if (storedTrack.track.mode === 'disabled') {...}` conditional so that the track mode is only updated if it is currently disabled. That way devs can still programmatically _enable_ metadata tracks during native iOS playback without us forcibly restoring them to their previous `disabled` state. Thoughts?